test/backend/test.py

```diff
@@ -45,177 +45,6 @@
-    def __init__(self, path, entry_point):
-        super().__init__(path, entry_point)
-
-    def is_input_le(self, inputs):
-        inputs_endianness = list(map(lambda x: x.dtype.byteorder, inputs))
-        endianness_is_consistent = len(set(inputs_endianness)) <= 1
-        assert endianness_is_consistent, \
-            "Input arrays contain a mixture of endianness configuration."
-
-        sys_is_le = sys.byteorder == 'little'
-        # To interpret character symbols indicating endianness:
-        # https://numpy.org/doc/stable/reference/generated/numpy.dtype.byteorder.html
-        explicitly_le = inputs_endianness[0] == "<"
-        implicitly_le = (inputs_endianness[0] == "=" and sys_is_le)
-        return explicitly_le or implicitly_le
-
-    def run(self, inputs, **kwargs):
-        if len(inputs):
-            # Deduce desired endianness of output from inputs.
-            sys_is_le = sys.byteorder == 'little'
-            inp_is_le = self.is_input_le(inputs)
-            if (sys_is_le != inp_is_le):
-                inputs = list(
-                    map(lambda x: x.byteswap().newbyteorder(), inputs))
-            outputs = super().run(inputs)
-            if (sys_is_le != inp_is_le):
-                outputs = list(
-                    map(lambda x: x.byteswap().newbyteorder(), outputs))
-            return outputs
-        else:
-            # Can't deduce desired output endianess, fingers crossed.
-            warnings.warn(
-                "Cannot deduce desired output endianness, using native endianness by default."
-            )
-            return super().run(inputs)
-
-
-class DummyBackend(onnx.backend.base.Backend):
-    @classmethod
-    def prepare(cls, model, device='CPU', **kwargs):
-        super(DummyBackend, cls).prepare(model, device, **kwargs)
-        name = model.graph.name
-        model_name = result_dir+name+".onnx"
-        exec_name = result_dir+name + ".so"
-        # Clean the temporary files in case
-        # Save model to disk as temp_model.onnx.
-        onnx.save(model, model_name)
-        if not os.path.exists(model_name) :
-            print("Failed save model: "+ name)
-
-        # Call frontend to process temp_model.onnx, bit code will be generated.
-        my_input, my_dim = determine_dynamic_parameters(name)
-        execute_commands([TEST_DRIVER, model_name], my_input, my_dim)
-        if not os.path.exists(exec_name) :
-            print("Failed " + test_config.TEST_DRIVER_COMMAND + ": " + name)
-        return EndiannessAwareExecutionSession(exec_name,
-                                               "run_main_graph")
-
-    @classmethod
-    def supports_device(cls, device):
-        d = Device(device)
-        if d.type == DeviceType.CPU:
-            return True
-        return False
-
-
-backend_test = onnx.backend.test.BackendTest(DummyBackend, __name__)
-
 # Test directories:
 # https://github.com/onnx/onnx/tree/master/onnx/backend/test/data/node
```

Comment:
Thanks for unifying the dict, much better to have one repo. Please add a quick comment on the format, presumably
{arg:{which dim to make unknown} }

Are the -1 allowed there? (just a question, not a request to add)

Reply:
Added a comment for this. Yes, -1 is allowed, meaning all inputs or all dimensions.